src/components/Dashboard/AccountBanner.js

```diff
@@ -176,9 +177,11 @@ const BannerWithProbability = ({ activeBalance }) => {
   // Calculate probability (since the total active balance is asynchronous
   // it can happen that it has not been updated yet when the juror active balance has)
   const draftingProbability = Math.min(1, totalPercentage / 100)
-
   const probablilityTooLow = totalPercentage < 1
-  const chances = totalPercentage > 0 && Math.floor(100 / totalPercentage)
+
+  const chances = probablilityTooLow
+    ? '100+'
+    : totalPercentage > 0 && Math.floor(100 / totalPercentage)
```

Comment:
Isn’t `totalPercentage > 0` always true if `probabilityTooLow` is `false`?

Reply:
oh, that is to avoid to divide by 0 given that `totalPercentage` is calculated async

Reply:
Perhaps we can restructure this component based on this loading state.

If it's still loading, we render `BannerLoadingRing`, otherwise, we render the `Wrapper` (and find a better name for it :smile:) and that component should handle all logic with the assumption that balances have already been loaded.

In the case of not wanting to divide by 0; we get `Infinity` back, and we can simply use:

```
const chances = Math.floor(100 / totalPercentage)
const chancesText = chances > 100 ? '100+' : chances
```

Reply:
not sure if i understood why we should change that is working now

Reply:
The reason I suggest a restructuring is that in the current `BannerWithProbability ` component, there is a lot of logic dependent on the asynchronously fetched variable, `totalActiveBalanceCurrentTerm`.

The resulting logic ([l172-184](https://github.com/aragon/court-dashboard/blob/ba3ee791a97fb4da87b9ac8e7ef1d86023d01e51/src/components/Dashboard/AccountBanner.js#L172-L184)) would be easier to reason about if we separated the component such that any logic reliant on `totalActiveBalanceCurrentTerm` could assume that it has been resolved.

For example, right now, it's unclear what happens if `getPercentageBN()` is passed `bn(-1)` as the denominator; I assume it handles it, but it puts this bit of code in a weird position in that there's an extra case to think through and be responsible for.

Moreover, this component's render is already in a good position to be logically restructured: we can immediately short circuit and render `BannerLoadingRing` when we know `totalActiveBalanceCurrentTerm` is still being fetched.

Reply:
Gotcha! thanks!